arrow/ipc/reader.go

```diff
@@ -36,12 +36,15 @@ import (
 // Reader expects a schema (plus any dictionaries) as the first messages
 // in the stream, followed by records.
 type Reader struct {
+	// refCount needs to be 64-bit aligned
+	// https://pkg.go.dev/sync/atomic#pkg-note-BUG
+	refCount int64
```

Comment:
does this actually ensure alignment? Or should we use atomic.Int64 ?

Reply:
It does because it's now at the start of the struct. The issue was that `schema *arrow.Schema` is a pointer, which is 32-bit on `arm` (without it being `arm64`). 

Using `atomic.Int64` would be indeed a safer option since it does [alignment automatically](https://cs.opensource.google/go/go/+/refs/tags/go1.24.1:src/sync/atomic/type.go;l=101) . Good point! :+1:

Reply:
I swapped with `atomic.Int64` now. I think this would need to be done on the complete code base though. If I have some time next week I can offer to do this. Along with that I could also double check that we don't have more alignment issues I did not stumble into yet.